src/github.com/matrix-org/dendrite/clientapi/readers/profile.go

```diff
@@ -175,13 +238,96 @@ func SetDisplayName(
 	}
 }
 
-func getLocalPart(userID string) string {
-	if !strings.HasPrefix(userID, "@") {
-		panic(fmt.Errorf("Invalid user ID"))
+func buildMembershipEvents(
+	memberships []authtypes.Membership, db *accounts.Database,
+	oldProfile *authtypes.Profile, changedKey string, newValue string,
+	userID string, cfg *config.Dendrite, queryAPI api.RoomserverQueryAPI,
+) ([]gomatrixserverlib.Event, error) {
+	ev := []gomatrixserverlib.Event{}
```

Comment:
I'd call "ev" something like "events" or "evs". Using plurals is a good way to hint to the reader that the variable is a collection rather than a single object.